torchgen/model.py

```diff
@@ -1045,12 +1179,26 @@ def parse(func: str) -> "FunctionSchema":
         assert str(r) == func, f"{str(r)} != {func}"
         return r
 
+    def returns_are_aliased(self) -> bool:
+        # We assert earlier that schemas can't have a mix of aliased and non-aliased returns
+        return any(
+            r
+            for r in self.returns
+            if r.annotation is not None and r.annotation.is_write
```

Comment:
If we make the returns more structured this can be a more trivial query (instead of having to do a scan over all returns). IDK if it's possible though, our functions have to be uniform enough.